pkg/apis/aws/types_backupbucket.go

```diff
@@ -0,0 +1,48 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package aws
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// RetentionType defines the level at which immutability properties are applied on objects.
+type RetentionType string
+
+const (
+	// BucketLevelImmutability sets the immutability feature on the bucket level.
+	BucketLevelImmutability RetentionType = "bucket"
+	// ComplianceMode is for "compliance" mode immutability.
+	ComplianceMode string = "compliance"
+	// GovernanceMode mode is of "governance" mode immutability.
+	GovernanceMode string = "governance"
```

Comment:
Maybe here it would fit if these were a separate type

Reply:
addressed. PTAL